.github/workflows/pid_controller_automated_test.yml

```diff
@@ -0,0 +1,57 @@
+---
+name: "PID Controller Automated Test"
+
+# yamllint disable-line rule:truthy
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+concurrency:
+  group: ${{ github.ref }}-pid-test
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+  pull-requests: write
+jobs:
+  pid-controller-automated-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt-get update && sudo apt-get install -y build-essential libssl-dev libmysqlclient-dev imagemagick libmagickwand-dev
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+      - uses: ruby/setup-ruby@0481980f17b760ef6bca5e8c55809102a0af1e5a # v1.263.0
+        with:
+          bundler-cache: true
+
+      - name: Setup Environment
+        run: |
+          git add .
```

Comment:
what's the purpose of those two lines? 🤔

Reply:
This was to clean the environment from previous failed checks. It just ensures the environment is completely clean. We need this to be able to successfully checkout new branch.